### ISSUE TITLE
[Question] doCreateItemEx

### ISSUE THREAD

**Issue by Riverlance** [otland/forgottenserver#2533]
Look at this function (is easier to understand what I will explain):

``` Lua
function addMeatExampleFunction()
  local item = Item(doCreateItemEx(2666, 7))
  if not item then return false end
  if player:addItemEx(item, false) == RETURNVALUE_NOERROR then -- (1)
    return true
  else
    item:remove() -- (2)
  end
  return false
end
```

(1) At this line, if this is false, means that was not possible to add the item to player.
**So, my question is, when this happens, is the (2) line necessary?**

I was looking about it and I found a function called `ScriptEnvironment::resetEnv (luascript.cpp)`.
And, as I understood, these tempItems that were created virtually on Lua and that was not removed, remains at the Lua Environment until you reload the whole Lua Environment.

Thanks in advance.

@djarek @WibbenZ @Mkalo @nekiro @TheSumm

**Comment by andersonfaaria** [comment]
you don't have to remove item, the garbage collector will do it in lua and the pointer will be reseted in c++ if it is not added

**Comment by Riverlance** [comment]
> you don't have to remove item, the garbage collector will do it in lua and the pointer will be reseted in c++ if it is not added

Yeah, you are right.

okk3 suggested me to print values and this was the result:

I putted a print inside ScriptEnvironment::resetEnv() (in which clear the tempItems list) and made a function to print the size of tempItems list. When I started the server, this happened:
https://i.gyazo.com/344511efd30da39ff3de645ef7916a49.mp4
(this function executes too fast)

Then I tried to create 1000 items without removing them.
![image](https://user-images.githubusercontent.com/2280432/53217779-2d719e00-3638-11e9-878c-8d335dcebb4e.png)
The first "0" value was printed before I create these items.
The "1000" value was after it.
The second "0" value was printed with the addEvent function with 1 ms of delay.

To me, this means it always clear the tempItems list as soon as possible.

Thanks for your attention.
